Let's answer three questions using the airports datast:

1) What is the northernmost airport in the United States?
2) What is the easternmost airport in the United States?
3) On February 12th, 2013, which New York area airport had the windiest weather?

In [12]:
import pandas as pd

df_airports = pd.read_csv('/home/kobi/is362/Week 5 Assignment/airports.csv')

In [13]:
# 1) Let's sort airports by latitude and pull the top airport to find the northernmost airport in the US
northernmost_airport = df_airports.sort_values(by='lat', ascending=False).head(1)
print("Northernmost Airport:")
display(northernmost_airport[['faa', 'name', 'lat', 'lon']])

Northernmost Airport:


faa                     name        lat        lon
417  EEN  Dillant Hopkins Airport  72.270833  42.898333

In [14]:
# 2) Let's sort airports by longitude and pull the easternmost airport in the US

easternmost_airport = df_airports.sort_values(by='lon', ascending=True).head(1)
print("Easternmost Airport:")
display(easternmost_airport[['faa', 'name', 'lat', 'lon']])

Easternmost Airport:


faa          name     lat      lon
95  ADK  Adak Airport  51.878 -176.646

In [15]:
# 3) Let's import the weather dataset, filter by New York airports, and then find the airport with highest wind speed

df_weather = pd.read_csv('/home/kobi/is362/Week 5 Assignment/weather.csv')

ny_weather = df_weather[(df_weather['origin'].isin(['JFK', 'LGA', 'EWR'])) & (df_weather['year'] == 2013) & (df_weather['month'] == 2) & (df_weather['day'] == 12)]

windiest_airport = ny_weather.sort_values(by='wind_speed', ascending=False).head(1)
print("Windiest New York Airport on February 12th, 2013:")
display(windiest_airport[['origin', 'wind_speed']])

Windiest New York Airport on February 12th, 2013:


origin  wind_speed
1009    EWR  1048.36058

Note: I consider EWR to be a new york "area" airport.

Issue #1- Dillant Hopkins Airport is in NH- that does not seem nearly as north as we can get! Let's show the top 5 to see if the top entry is a mistake!

In [16]:
northernmost_airport = df_airports.sort_values(by='lat', ascending=False).head(5)
print("Northernmost Airports:")
display(northernmost_airport[['faa', 'name', 'lat', 'lon']])

Northernmost Airports:


faa                                        name        lat         lon
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003
110  AIN                          Wainwright Airport  70.638056 -159.994722
708  K03                               Wainwright As  70.613378 -159.860350
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000

Seems that the latitude and longitude of Dillant Hopkins Airport were flipped in the dataset! It seems that Wiley Post Will Rogers Memorial airport in ALASKA (!) is the TRUE northmost airport. The rest of the entries seem to also be correct.

Issue #2- Adak Airport SEEMS to be the westmost airport... let's show the top 5 eastmost airports to see if this is a mistake...

In [17]:
easternmost_airport = df_airports.sort_values(by='lon', ascending=True).head(5)
print("Easternmost Airports:")
display(easternmost_airport[['faa', 'name', 'lat', 'lon']])

Easternmost Airports:


faa              name        lat         lon
95    ADK      Adak Airport  51.878000 -176.646000
112   AKB      Atka Airport  52.220278 -174.206389
522   GAM   Gambell Airport  63.766800 -171.733000
1281  SVA  Savoonga Airport  63.686400 -170.493000
1251  SNP    St Paul Island  57.167333 -170.220444

Hmm... let's think about these negative longitudes... Alaska just so happens to be a state that the 180th meridian crosses through. This makes Adak airport the easternmost US airport in the... WESTERN hemisphere. Since the earth is a globe, this would make Adak airport the easternmost airport in the United States based off this definition.

Issue #3- What if there are outliers in the wind dataset? A wind speed of 1048 mph or kmph seems quite fast... as the rest of the data has more normal looking values... Let's display the top 10!

In [18]:
windiest_airport = ny_weather.sort_values(by='wind_speed', ascending=False).head(10)
print("Windiest New York Airport on February 12th, 2013:")
display(windiest_airport[['origin', 'wind_speed']])

Windiest New York Airport on February 12th, 2013:


origin  wind_speed
1009     EWR  1048.36058
18417    LGA    23.01560
1018     EWR    21.86482
18428    LGA    21.86482
9717     JFK    20.71404
1008     EWR    20.71404
18429    LGA    20.71404
18416    LGA    19.56326
9716     JFK    19.56326
18427    LGA    19.56326

BANG BANG! Does seem like the EWR wind speed is an outlier. Seems LGA was the windiest! 

BUT WAIT! This is only showing us the highest single hour value. Taking a deeper look into the dataset shows that there are recordings for every hour of every day. We should find the average wind speeds of Feb 12, 2013 in each of the three airports to compare.

In [20]:
ny_airports_weather = df_weather[(df_weather['origin'].isin(['LGA', 'EWR', 'JFK'])) & 
                                 (df_weather['year'] == 2013) & 
                                 (df_weather['month'] == 2) & 
                                 (df_weather['day'] == 12)]

# Remove the EWR outlier

ny_airports_weather = ny_airports_weather[ny_airports_weather['wind_speed'] != 1048.36058]\

# Avg wind speeds grouped by airport

average_wind_speeds = ny_airports_weather.groupby('origin')['wind_speed'].mean()

# Display results

print("Average Wind Speeds on February 12, 2013:")
display(average_wind_speeds)

Average Wind Speeds on February 12, 2013:


origin
EWR    13.258987
JFK    14.384750
LGA    14.960140
Name: wind_speed, dtype: float64

Conclusions:

Wiley Post Will Rogers Memorial Airport is the northernmost airport in the US.

By a technical definition, Adak Airport is the "eastern" most airport in the US.

LGA had the highest SINGLE HOUR and AVERAGE wind speeds on February 12, 2013.